### PR TITLE
CDS-34265-skip-label-selector-tracking-for-k8s-deployments

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
Added a new topic to the K8s deployment section under CD-Advanced

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by abosii-harness — DISMISSED** [review]

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+You can prevent Harness from using its default Kubernetes label selector `harness.io/track: stable` during canary deployments. Skipping this label can help when you have existing non-Harness deployed services or conflicts in naming. Currently, this feature is behind the feature flag `SKIP_ADDING_TRACK_LABEL_SELECTOR_IN_ROLLING`. Contact [Harness Support](mailto:support@harness.io) to enable the feature. 
+
+### Review: Labels in Harness Kubernetes canary phases
+
+Default Harness Kubernetes canary deployments are a two phase process that relies on deployment object labels for tracking. 
+
+1. **Canary Phase:**
+	1. Harness creates a canary version of the Kubernetes deployment object defined in your service definition **Manifests** section.
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:14]: Canary phase maybe?

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	2. All pods in the canary phase have the label `harness.io/track: canary`. Traffic could be routed to these pods using this label.
+	3. Once that deployment is verified, the canary delete step deletes it by default.    
+	Using this method, Harness provides a canary group as a way to test the new build, run your verification, and then roll out to the following primary group.
+2. **Primary Phase:**
+	1. Runs the actual deployment using a Kubernetes rolling update with the number of pods you specify in the **Manifests** files. For example, `replicas: 3`.
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:19]: Primary phase

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+title: Skip Harness Label Selector Tracking on Kubernetes Deployments
+description: Prevent Harness from using its default Kubernetes label selector harness.io/track --  stable during canary deployments.
+sidebar_position: 9
+---
+
+You can prevent Harness from using its default Kubernetes label selector `harness.io/track: stable` during canary deployments. Skipping this label can help when you have existing non-Harness deployed services or conflicts in naming. Currently, this feature is behind the feature flag `SKIP_ADDING_TRACK_LABEL_SELECTOR_IN_ROLLING`. Contact [Harness Support](mailto:support@harness.io) to enable the feature. 
+
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:8]: You can prevent Harness from using its default Kubernetes label selector, `harness.io/track: stable` during canary deployments. Skipping this label can help when you have existing non-Harness deployed services or naming conflicts. Currently, this feature is behind the feature flag `SKIP_ADDING_TRACK_LABEL_SELECTOR_IN_ROLLING`.

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+### Review: Labels in Harness Kubernetes canary phases
+
+Default Harness Kubernetes canary deployments are a two phase process that relies on deployment object labels for tracking. 
+
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:12]: Default Harness Kubernetes canary deployments are a two-phase process that relies on deployment object labels for tracking.

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+1. **Canary Phase:**
+	1. Harness creates a canary version of the Kubernetes deployment object defined in your service definition **Manifests** section.
+	2. All pods in the canary phase have the label `harness.io/track: canary`. Traffic could be routed to these pods using this label.
+	3. Once that deployment is verified, the canary delete step deletes it by default.
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:16]: I didn't understand: canary delete step deletes what?

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+### Review: Labels in Harness Kubernetes canary phases
+
+Default Harness Kubernetes canary deployments are a two phase process that relies on deployment object labels for tracking. 
+
+1. **Canary Phase:**
+	1. Harness creates a canary version of the Kubernetes deployment object defined in your service definition **Manifests** section.
+	2. All pods in the canary phase have the label `harness.io/track: canary`. Traffic could be routed to these pods using this label.
+	3. Once that deployment is verified, the canary delete step deletes it by default.    
+	Using this method, Harness provides a canary group as a way to test the new build, run your verification, and then roll out to the following primary group.
+2. **Primary Phase:**
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:18]: Harness provides a canary group as a way to test the new build, run your verification, and roll out the canary group to the following primary group using this method.
Is this statement still correct after I rephrase? Can we rephrase this?

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+
+```
+The Deployment “harness-example-deployment” is invalid: spec.selector:   
+  Invalid value: v1.LabelSelector{MatchLabels:map[string]string{“app”:“harness-example”},   
+  MatchExpressions:[]v1.LabelSelectorRequirement{}}: field is immutable
+```
+
+Most often, you can delete or rename the deployment object. In some case, you will experience downtime when the object restarts. 
+
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:38]: In some cases, you will experience downtime when the object restarts.

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+As an alternative, you can force Harness to skip the `harness.io/track: stable` label in the canary deployment.
+
+### Skipping the tracking label
+
+Once the relevant feature flag is enabled, a Harness Kubernetes canary deployment works like this:
+
+* If the deployment object already exists in the cluster without the `harness.io/track: stable`, Harness will not add the `harness.io/track: stable` label to the deployment object.
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:45]: If the deployment object already exists in the cluster without the `harness.io/track: stable` label, ...

**Review by SmritiSatya — DISMISSED** [review]

**Review by SmritiSatya — APPROVED** [review]

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
+---
+title: Skip Harness Label Selector Tracking on Kubernetes Deployments
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:2]:
```suggestion
title: Skip Harness label selector tracking on Kubernetes deployments
```

Comment (michaelcretzman) [reply docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:2]: titles should be in sentence case: https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+style+guide

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+description: Prevent Harness from using its default Kubernetes label selector during canary deployments.
+sidebar_position: 9
+---
+
+You can prevent Harness from using its default Kubernetes label selector, `harness.io/track: stable` during canary deployments. Skipping this label can help when you have existing non-Harness deployed services or naming conflicts. Currently, this feature is behind the feature flag `SKIP_ADDING_TRACK_LABEL_SELECTOR_IN_ROLLING`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:7]:
```suggestion
:::note
Currently, this feature is behind the feature flag `SKIP_ADDING_TRACK_LABEL_SELECTOR_IN_ROLLING`. Contact [Harness Support](mailto:support@harness.io) to enable the feature. 
:::

You can prevent Harness from using its default Kubernetes label selector, `harness.io/track: stable`, during canary deployments. Skipping this label can help when you have existing, non-Harness deployed services or naming conflicts. 
```

Comment (michaelcretzman) [reply docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:7]: Feature flag notes go at the top of the topic.

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+### Review: Labels in Harness Kubernetes canary phases
+
+Default Harness Kubernetes canary deployments are a two-phase process that relies on deployment object labels for tracking. 
+
+1. **Canary Phase**
+	1. Harness creates a canary version of the Kubernetes deployment object defined in your service definition **Manifests** section.
+	2. All pods in the canary phase have the label `harness.io/track: canary`. Traffic could be routed to these pods using this label.
+	3. Once this deployment is verified, the canary delete step deletes the deployment by default.    
+   
+	Using this method, Harness provides a canary group as a way to test the new build, run your verification, and roll out the canary group to the following primary group.
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:18]:
```suggestion
	Using this method, Harness provides a canary pod set as a way to test the new build, run your verification.
```

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+2. **Primary Phase**
+	1. Runs the actual deployment using a Kubernetes rolling update with the number of pods you specify in the **Manifests** files. For example, `replicas: 3`.
+	2. All pods in the primary phase have the label `harness.io/track: stable`.
+
+### Review: Invalid value LabelSelector
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:23]:
```suggestion
## Invalid value LabelSelector
```

Comment (michaelcretzman) [reply docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:23]: we don't use `Review` or `Step` or `Option` in headings anymore. We have a lot left from before, but moving forward we don't use them.

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
+---
+title: Skip Harness Label Selector Tracking on Kubernetes Deployments
+description: Prevent Harness from using its default Kubernetes label selector during canary deployments.
+sidebar_position: 9
+---
+
+You can prevent Harness from using its default Kubernetes label selector, `harness.io/track: stable` during canary deployments. Skipping this label can help when you have existing non-Harness deployed services or naming conflicts. Currently, this feature is behind the feature flag `SKIP_ADDING_TRACK_LABEL_SELECTOR_IN_ROLLING`. Contact [Harness Support](mailto:support@harness.io) to enable the feature. 
+
+### Review: Labels in Harness Kubernetes canary phases
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:9]:
```suggestion
## Labels in Harness Kubernetes canary phases
```

File: docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+The Deployment “harness-example-deployment” is invalid: spec.selector:   
+  Invalid value: v1.LabelSelector{MatchLabels:map[string]string{“app”:“harness-example”},   
+  MatchExpressions:[]v1.LabelSelectorRequirement{}}: field is immutable
+```
+
+Most often, you can delete or rename the deployment object. In some cases, you will experience downtime when the object restarts. 
+
+As an alternative, you can force Harness to skip the `harness.io/track: stable` label in the canary deployment.
+
+### Skipping the tracking label
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/cd-kubernetes-category/skip-harness-label-selector-tracking-on-kubernetes-deployments.md:42]:
```suggestion
## Skipping the tracking label
```

**Review by michaelcretzman — APPROVED** [review]
approving with the assumption that you will consider my suggestions.

**Review by charanya-j — APPROVED** [review]

**Pull request merged**